src/components/Menu/Menu.tsx

```diff
@@ -49,33 +55,60 @@ export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
  *
  * A dropdown that reveals or hides a list of actions
  */
-export const Menu = ({ className, ...other }: MenuProps) => {
+export const Menu = ({
+  buttonText,
+  children,
+  className,
+  ...other
+}: MenuProps) => {
   const menuClassNames = clsx(className, styles['menu']);
-  return <HeadlessMenu as="div" className={menuClassNames} {...other} />;
+  if (typeof children === 'function') {
+    return (
+      <HeadlessMenu as="div" className={menuClassNames} {...other}>
+        {({ open, close }) => (
+          <>
+            {buttonText && (
+              <MenuButton className={styles['menu__button']}>
+                {buttonText}
+                <Icon
+                  className={styles['menu__button--icon']}
+                  name="expand-more"
+                  purpose="decorative"
+                  size="1.25rem"
+                />
+              </MenuButton>
+            )}
+            {children({ open, close })}
```

Comment:
Would prefer to remove `close` here since `Menu` closes on click anyways but TS complains if `close` is not passed
![image](https://user-images.githubusercontent.com/86632227/221769229-14d54eea-bc4f-4e9a-8e8f-44be5af7bcb0.png)

Reply:
With alt approach, no longer a concern